java/src/org/openqa/selenium/bidi/BiDiProvider.java

```diff
@@ -44,18 +44,32 @@ public Class<HasBiDi> getDescribedInterface() {
 
   @Override
   public HasBiDi getImplementation(Capabilities caps, ExecuteMethod executeMethod) {
+    return new HasBiDi() {
```

Comment:
I am not 100% sure about the BiDi part because the user is expressing the intention to use BiDi by setting `webSocketUrl: true`. I think it makes sense for CDP because we are doing that implicitly.

Reply:
I get that, but if the user has set the capability, then it means they want to use BiDi methods. I would prefer to avoid this change. The CDP one seems fine.